### PR TITLE
Workaround for has_size Cassette issue

### PR DESCRIPTION
**Pull request by mateuszbaran**
Reported here: https://github.com/JuliaArrays/StaticArrays.jl/issues/1047

Ref: https://github.com/JuliaLabs/Cassette.jl/issues/198

### CODE REVIEW

**Comment by mateuszbaran** [comment]
@N5N3 could you review this change (you've changed that code in #1016)?

**Review by N5N3 — APPROVED** [review]

File: src/convert.jl [hunk 1]
```diff
@@ -19,6 +19,9 @@ end
 @pure has_eltype(::Type{<:StaticArray}) = false
 @pure has_size(::Type{<:StaticArray{S}}) where {S<:Tuple} = @isdefined S
 @pure has_size(::Type{<:StaticArray}) = false
+# workaround for https://github.com/JuliaArrays/StaticArrays.jl/issues/1047
+@pure has_size(::Type{SVector}) = false
```
[the remark this answers is not in the document]
Comment (N5N3) [reply src/convert.jl:23]: Then I think there's no need to block this.
But it should be good to cover the cases we already found?
```julia
for T in (:S, :M)
   @eval has_size(::Type{$(Symbol(T,:Vector))}) = false
   @eval has_size(::Type{$(Symbol(T,:Matrix))}) = false
   @eval has_size(::Type{$(Symbol(T,:Matrix)){N}}) where {N} = false
end
```

[the remark this answers is not in the document]
Comment (mateuszbaran) [reply src/convert.jl:23]: Sure, we can have that.

File: src/convert.jl [hunk 2]
```diff
@@ -20,7 +20,11 @@ end
 @pure has_size(::Type{<:StaticArray{S}}) where {S<:Tuple} = @isdefined S
 @pure has_size(::Type{<:StaticArray}) = false
 # workaround for https://github.com/JuliaArrays/StaticArrays.jl/issues/1047
-@pure has_size(::Type{SVector}) = false
+for T in (:S, :M)
+    @eval has_size(::Type{$(Symbol(T,:Vector))}) = false
+    @eval has_size(::Type{$(Symbol(T,:Matrix))}) = false
+    @eval has_size(::Type{$(Symbol(T,:Matrix)){N}}) where {N} = false
+ end
```
Comment (hyrodium) [inline src/convert.jl:27]:
```suggestion
for T in (:S, :M)
    @eval has_size(::Type{$(Symbol(T,:Vector))}) = false
    @eval has_size(::Type{$(Symbol(T,:Matrix))}) = false
    @eval has_size(::Type{$(Symbol(T,:Matrix)){N}}) where {N} = false
end
```

Alternatively, I think we can remove the for loop.

```suggestion
has_size(::Type{SVector}) = false
has_size(::Type{MVector}) = false
has_size(::Type{SMatrix}) = false
has_size(::Type{MMatrix}) = false
has_size(::Type{SMatrix{N}}) where {N} = false
has_size(::Type{MMatrix{N}}) where {N} = false
```

**Comment by ferrolho** [comment]
Would it be okay to tag a new release to include this fix?

**Pull request merged**